### PR TITLE
Make LSI dispatcher CLI param for number of jobs optional

### LINKED ISSUE #3104: lsi_dispatcher is not working from command-line when not specifying maxsize argument
#### Problem description

When running `lsi_dispatcher` from the command-line, if you don't specify the `maxsize` argument explicitly, you get an error for the missing positional argument:

```
usage: lsi_dispatcher.py [-h] maxsize
lsi_dispatcher.py: error: the following arguments are required: maxsize
```

According to the documentation, this argument should be optional.

The issue seems to be that the nargs argument to `add_argument` is missing:

```python
    parser.add_argument(
        'maxsize', type=int, help='Maximum number of jobs to be kept pre-fetched in the queue.', default=MAX_JOBS_QUEUE
    )
 ```
In order to make this argument optional, this should be:

```python
    parser.add_argument(
        'maxsize', nargs='?', type=int, help='Maximum number of jobs to be kept pre-fetched in the queue.', default=MAX_JOBS_QUEUE
    )
```

#### Steps/code/corpus to reproduce

Include full tracebacks, logs and datasets if necessary. Please keep the examples minimal ("minimal reproducible example").

If your problem is with a specific Gensim model (word2vec, lsimodel, doc2vec, fasttext, ldamodel etc), include the following:

```python
$ python3 -m gensim.models.lsi_dispatcher
usage: lsi_dispatcher.py [-h] maxsize
lsi_dispatcher.py: error: the following arguments are required: maxsize
```

#### Versions

```python
Linux-5.4.0-67-generic-x86_64-with-glibc2.2
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0]
Bits 64
NumPy 1.19.4
SciPy 1.6.0
gensim 4.0.1
FAST_VERSION 1
```

### PR DESCRIPTION
**Pull request by robguinness**
Closes #3104.

### CODE REVIEW

**Comment by mpenkov** [comment]
CI failure appears to be unrelated to the PR. I'm rerunning the failed build just to be sure.

File: gensim/models/lsi_dispatcher.py [hunk 1]
```diff
@@ -278,7 +278,11 @@ def exit(self):
     logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
     parser = argparse.ArgumentParser(description=__doc__[:-135], formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument(
-        'maxsize', type=int, help='Maximum number of jobs to be kept pre-fetched in the queue.', default=MAX_JOBS_QUEUE
+        'maxsize',
+        nargs='?',
+        type=int,
+        help='Maximum number of jobs to be kept pre-fetched in the queue.',
+        default=MAX_JOBS_QUEUE
```
Comment (mpenkov) [inline gensim/models/lsi_dispatcher.py:285]:
```suggestion
        default=MAX_JOBS_QUEUE,
```

**Comment by mpenkov** [comment]
CI failure appears to be unrelated to the PR. I'm rerunning the failed build just to be sure.

@robguinness Can you please make this PR editable to maintaners? See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork.

**Comment by robguinness** [comment]
> CI failure appears to be unrelated to the PR. I'm rerunning the failed build just to be sure.
> 
> @robguinness Can you please make this PR editable to maintaners? See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork.

Hi @mpenkov, I don't actually see that option to make the PR editable. But I can push that change.

**Comment by robguinness** [comment]
Commit has been added.

**Comment by mpenkov** [comment]
OK, merged. Thank you @robguinness !

**Pull request merged**